XCTestBootstrapTests/FBTestManagerTestReporterJUnitTests.m

```diff
@@ -0,0 +1,111 @@
+                                                                   finishingAt:finishTime                              \
+                                                                      runCount:@(runCountValue)                        \
+                                                                      failures:@(failureCount)                         \
+                                                                    unexpected:@(unexpectedCount)                      \
+                                                                  testDuration:@(testTime)                             \
+                                                                 totalDuration:@(totalTime)]]
+
+#define assertJUnitReportEqualTo(file, extension)                                                                      \
+  [self.reporter testManagerMediatorDidFinishExecutingTestPlan:self.testManagerAPIMediator];                           \
+  NSError *error;                                                                                                      \
+  NSString *path =                                                                                                     \
+      [[NSBundle bundleForClass:self.class] pathForResource:@"expectedJUnitReporterResult" ofType:@"xml"];             \
+  NSString *expected = [NSString stringWithContentsOfFile:path encoding:NSUTF8StringEncoding error:&error];            \
+  XCTAssertNil(error);                                                                                                 \
+  NSString *actual =                                                                                                   \
+      [[NSString alloc] initWithContentsOfFile:self.outputFilePath encoding:NSUTF8StringEncoding error:&error];        \
+  XCTAssertNil(error);                                                                                                 \
+  XCTAssertEqualObjects(expected, actual);
+
+@interface FBTestManagerTestReporterJUnitTests : XCTestCase
+
+@property (nonatomic) id testManagerAPIMediator;
+@property (nonatomic) FBTestManagerTestReporterJUnit *reporter;
+@property (nonatomic) NSFileHandle *outputFileHandle;
+@property (nonatomic) NSString *outputFilePath;
+@property (nonatomic, readonly) NSString *outputFileContent;
+
+@end
+
+@implementation FBTestManagerTestReporterJUnitTests
+
+- (void)setUp
+{
+  [super setUp];
+
+  self.testManagerAPIMediator = [OCMockObject mockForClass:[FBTestManagerAPIMediator class]];
+  self.outputFilePath = [NSTemporaryDirectory() stringByAppendingPathComponent:[NSUUID UUID].UUIDString];
+
+  [[NSFileManager defaultManager] createFileAtPath:self.outputFilePath contents:nil attributes:nil];
+
+  self.outputFileHandle = [NSFileHandle fileHandleForWritingAtPath:self.outputFilePath];
+  self.reporter = [FBTestManagerTestReporterJUnit withOutputFileHandle:self.outputFileHandle];
```

Comment:
It looks like you never close `outputFileHandle` which means you'll leak file descriptors in these tests.